database.py

```diff
@@ -0,0 +1,178 @@
+from neo4j import GraphDatabase, exceptions, basic_auth
+from neo4j.exceptions import ServiceUnavailable
+from abc import ABC, abstractmethod
+from exceptions import PageNotFound, PostNotFound
+import logging
+import hashlib
+import datetime
+
+date = datetime.datetime.now()
+ts = date.timestamp()
+
+
+class Nodes(ABC):
+
+    @abstractmethod
+    def _create_page_node(self, tx, name):
+        pass
+
+    @abstractmethod
+    def _create_relationship(self, tx, page_name, time):
+        pass
+
+    @abstractmethod
+    def _create_post(self, tx, post_text, image, time):
+        pass
+
+    @abstractmethod
+    def _find_page(self, tx, page_name):
+        pass
+
+    @abstractmethod
+    def _find_post(self, tx, page_name):
+        pass
+
+    def _hash_everything(self, text, time):
+        pass
+
+
+class DataBase(Nodes):
+
+    def __init__(self, uri, user, password):
+        self.uri = uri
+        self.user = user
+        self.password = password
+        self.driver = None
+        try:
+            self.driver = GraphDatabase.driver(uri, auth=basic_auth(user, password))
+        except Exception(exceptions.AuthError):
+            raise
+
+    def close(self):
+        if self.driver is not None:
+            self.driver.close()
+        else:
+            raise exceptions.DriverError
+
+    def check_connection(self):
+        self.driver.verify_connectivity()
+
+    def _create_page_node(self, tx, name):
+        try:
+            return tx.run("CREATE (a:Page {Name: $name})", name=name)
+        except Exception(exceptions.CypherSyntaxError):
+            raise
+
+    def _create_relationship(self, tx, page_name, time):
+        query = (
+            "MATCH (a: Page), (b: Post) "
+            "WHERE a.Name = $name AND b.Time = $time "
+            "CREATE (a)-[t:Date {Time:$time}]->(b) "
+            "RETURN t"
+        )
+        result = tx.run(query, name=page_name, time=time)
+        try:
+            return result
+        except ServiceUnavailable as exception:
+            logging.error(f"{query} raised an error: \n {exception}")
+            raise
+
+    def _create_post(self, tx, post_text, image, time):
+        hashed_dict = self._hash_everything(post_text, time)
+        hashed_text = hashed_dict['text'].hexdigest()
+        hashed_time = hashed_dict['time'].hexdigest()
```

Comment:
_hash_everything should do this 

Basically here you do

hash = self._hash_everything. 

Also no need to seperate text and time I'll tell you more in the method